### PR TITLE
chore: Update dependencies to allow adding benchmarking

### PR DESCRIPTION
**Pull request by ulrikstrid**
## Problem

We need janestreet 0.15 packages to allow adding benchmarking dependencies

## Solution

Bump flakes and update esy

### CODE REVIEW

**Comment by ulrikstrid** [comment]
`esy` is going to be a pita to update, I don't have time to do that now, marking as draft

File: nix/shell.nix [hunk 1]
```diff
@@ -30,8 +30,8 @@ pkgs.mkShell {
       ocaml
       findlib
       dune_2
-      ocaml-lsp
-      ocamlformat-rpc
+      # ocaml-lsp
```
Comment (ulrikstrid) [inline nix/shell.nix:33]: Something broke this for OCaml 5.00

File: flake.nix [hunk 2]
```diff
@@ -46,21 +73,22 @@
 
         deku = pkgs.callPackage ./nix/deku.nix {
           doCheck = true;
-          nodejs = pkgs.nodejs-12_x;
+          nodejs = pkgs.nodejs-16_x;
```
Comment (ulrikstrid) [inline flake.nix:76]: There are a couple of these, that just sets it to the latest LTS release

File: nix/overlay.nix [hunk 3]
```diff
@@ -15,11 +15,6 @@ in {
       });
       dream = osuper.dream.overrideAttrs (o: { doCheck = false; });
       alcotest = osuper.alcotest.overrideAttrs (o: {
```
Comment (ulrikstrid) [inline nix/overlay.nix:17]: `alcotest` is 1.5.0 in `nixpkgs` now

File: nix/overlay.nix [hunk 4]
```diff
@@ -47,16 +42,6 @@ in {
           sha256 = "13c53b1cxkq2nj444655skw5a1mcxzbaqwqsqjf7jbwradb3hmxa";
         };
       });
-      odoc = osuper.odoc.overrideAttrs (o: {
```
Comment (ulrikstrid) [inline nix/overlay.nix:50]: Fixed upstream

File: src/bin/deku_node.ml [hunk 5]
```diff
@@ -141,8 +141,7 @@ let node folder prometheus_port =
              handle_withdraw_proof;
              handle_ticket_balance;
              handle_trusted_validators_membership;
-           ]
-      @@ Dream.not_found;
```
Comment (ulrikstrid) [inline src/bin/deku_node.ml:145]: Breaking change in Dream

File: .github/workflows/nix.yml [hunk 6]
```diff
@@ -15,8 +15,9 @@ jobs:
         with:
           name: deku
           authToken: "${{ secrets.CACHIX_SIGNING_KEY }}"
-      - name: "Run nix flake check"
-        run: nix flake check
+      # Broken because we patch nixpkgs, can add it back when janestreet 0.15 is added upstream
+      # - name: "Run nix flake check"
+      #   run: nix flake check
```
Comment (aguillon) [inline .github/workflows/nix.yml:20]: What does this change to the CI?

**Review by d4hines — APPROVED** [review]
Thanks for this @ulrikstrid. LGTM

**Pull request merged**